### PR TITLE
Flagutil: utility for json file from directory or url and unmarshal to struct

### PR DESCRIPTION
**Pull request by james-prysm**
**What type of PR is this?**

Feature

used in #10312 

**What does this PR do? Why is it needed?**
this PR adds a new utility for developers to send a string representing the file location or url and unmarshal it into a struct.
this will be important for features like fee recipient which takes in a file or url.

### CODE REVIEW

**Comment by kasey** [comment]
I have little feedback on how the functions themselves are implemented, solid work. High level feedback is, do we need a library for this, or should this code be specific to the fee recipient feature? These methods are combinations of fairly straightforward pieces of the standard library, so I'm not sure if a dedicated package is justified.

**Comment by james-prysm** [comment]
> I have little feedback on how the functions themselves are implemented, solid work. High level feedback is, do we need a library for this, or should this code be specific to the fee recipient feature? These methods are combinations of fairly straightforward pieces of the standard library, so I'm not sure if a dedicated package is justified.

that's fair, i was thinking that we might have similar situations ( like web3signer) that does this either parsing string or using a url. right now the fee recipient stuff is mixed into the validator struct etc I was thinking of just moving utility functions somewhere else. open to thoughts.

File: cmd/flagutil/jsontools.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+)
+
+func UnmarshalFromFileOrURL(ctx context.Context, from string, to interface{}) error {
```
[the remark this answers is not in the document]
Comment (kasey) [reply cmd/flagutil/jsontools.go]: This is definitely a matter of opinion, so take it with a grain of salt. Preferably a grain of salt on top of a chocolate chip cookie.

File: validator/node/node.go [hunk 2]
```diff
@@ -645,3 +648,65 @@ func clearDB(ctx context.Context, dataDir string, force bool) error {
 
 	return nil
 }
+
+func unmarshalFromURL(ctx context.Context, from string, to interface{}) error {
```
[the remark this answers is not in the document]
Comment (kasey) [reply validator/node/node.go:652]: and for just a bit more context, here is how the FlagOptions are set up and passed in to the node:
https://github.com/prysmaticlabs/prysm/blob/develop/cmd/beacon-chain/main.go#L234-L246
variadic argument on node.New: https://github.com/prysmaticlabs/prysm/blob/develop/beacon-chain/node/node.go#L108
applying the opts: https://github.com/prysmaticlabs/prysm/blob/develop/beacon-chain/node/node.go#L149
opt definitions: https://github.com/prysmaticlabs/prysm/blob/develop/beacon-chain/node/options.go

[the remark this answers is not in the document]
Comment (james-prysm) [reply validator/node/node.go:652]: will keep as-is for now but create an issue for tracking this

[the remark this answers is not in the document]
Comment (kasey) [reply validator/node/node.go:652]: In Discord we discussed addressing this in a follow-up, so I'm going to approve what you have here and we can clean up validator cmd initialization at another time. Added an issue to track it here: https://github.com/prysmaticlabs/prysm/issues/10339

**Review by kasey — APPROVED** [review]

**Pull request merged**